signer/src/stacks/api.rs

```diff
@@ -646,6 +631,82 @@ impl TryFrom<AccountEntryResponse> for AccountInfo {
     }
 }
 
+/// Minimal model type representing an epoch entry in a `/v2/pox` response,
+/// including only fields which we currently use.
+///
+/// Specifically, we do *not* use the `StacksEpochId` enum type from stacks-core
+/// as it would break if Stacks introduces new epochs prior to our dependencies
+/// being updated.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+struct PoxEpoch {
+    /// String representation of the epoch ID, e.g. `Epoch11`, `Epoch30`, `Epoch33`, etc.
+    epoch_id: String,
+    /// The Bitcoin block height at which this epoch activates.
+    start_height: BitcoinBlockHeight,
+}
+
+/// Minimal response type for the `/v2/pox` endpoint, including only fields
+/// which we currently use.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+struct PoxResponse {
+    /// The current Bitcoin block height, as known by the Stacks node based on
+    /// its current Stacks tenure. Note that if the Stacks node is behind, this
+    /// may be lower than the actual current Bitcoin block height.
+    current_burnchain_block_height: BitcoinBlockHeight,
+    /// The list of all known epochs, including their start heights. Used
+    /// primarily to determine the start height of Stacks epoch 3.0 (Nakamoto).
+    epochs: Vec<PoxEpoch>,
+}
+
+/// Information regarding whether or not we are in pre- or post-Nakamoto
+/// era.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StacksEpochStatus {
+    /// We are in the pre-Nakamoto era.
+    PreNakamoto {
+        /// The current bitcoin block height.
+        current_bitcoin_height: BitcoinBlockHeight,
```

Comment:
We might want to rename this to something like `stacks_node_bitcoin_height` (or whatever) to reflect that it is the current processed height of the stacks node. Right now it reads as though it is the current block height of the canonical bitcoin blockchain. We should also update the comment to be similar to the `StacksEpochStatus.current_burnchain_block_height` field.

Reply:
[004ff91](https://github.com/stacks-sbtc/sbtc/pull/1836/commits/004ff9160911c70cda109b27af9129a8e8ffd9ab)

Went with `reported_bitcoin_height` (just because `stacks_node_bitcoin_height` felt pretty long-winded), and updated the comment to match.